marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/builtins/steps/mapping/entity-services/lib.sjs

```diff
@@ -511,6 +526,21 @@ function validateAndTestUriExpressions(mapping, validatedMappingsArray, sourceIn
   return mapping;
 }
 
+function calculateInstanceCount(mapping, validatedMappingsArray, sourceInstance){
+  const namespaces = getMappingNamespacesObject(mapping);
+  validatedMappingsArray.forEach((entityMapping, mappingIndex) => {
+    if(mappingIndex != 0){
+      const sourceDocument = fn.head(xdmp.unquote(xdmp.quote(sourceInstance)));
```

Comment:
Sorry just noticed this - it looks like this should be outside the for loop so it's only calculated once, as it doesn't depend on what we're iterating over.